hazelcast/src/test/java/com/hazelcast/xa/HazelcastXATest.java

```diff
@@ -218,18 +219,19 @@ public void testParallel() throws Exception {
         ExecutorService executorService = Executors.newFixedThreadPool(5);
         final CountDownLatch latch = new CountDownLatch(size);
         for (int i = 0; i < size; i++) {
-            executorService.execute(new Runnable() {
-                public void run() {
-                    try {
-                        txn(instance);
-                    } catch (Exception e) {
-                        LOGGER.severe("Exception during txn", e);
-                    } finally {
-                        latch.countDown();
-                    }
+            executorService.execute(() -> {
+                try {
+                    txn(instance);
+                } catch (Exception e) {
+                    LOGGER.severe("Exception during txn", e);
+                } finally {
+                    latch.countDown();
                 }
             });
         }
+        // Wait for the executorService to finish all tasks
+        executorService.shutdown();
+        executorService.awaitTermination(20,TimeUnit.SECONDS);
         assertOpenEventually(latch, 20);
```

Comment:
I am probably missing something. This seems like we just moved the 20 s wait:
- we first shutdown the executor service and way for 20s to finish
- then check the latch (if the service is finished this will now not wait)

How is this changing anything? The `assertOpenEventually` already waits for ~20s

Reply:
Normally in a unit test, we have Arrange, Act, Assert steps. When we come to assert step we don't want anything else to be working. Maybe the test is going to fail again, but at least at assert state we will know that nothing else is working at the background.

Reply:
My point is that the new code will wait 20 s, if the work isn't done yet it will return false and the assert will fail, nothing really changed - we didn't increase the timeout, increase logging, no new log statements...

Reply:
There is nothing wrong with the test. It is failing because the waiting time was not sufficient. I have increased the waiting time to 60. If it could only finish 40 tasks in 20 seconds according to test output, 60 should be sufficient for 100 tasks